### PR TITLE
Added a get_manager method that returns manager by its type or subtype

### PR DESCRIPTION
**Pull request by DrInfy**
Added a method from getting managers by their type from the `Knowledge` class.

Attempts to solve #54 with as few breaking changes as possible and having proper generic type return from requesting the manager by its type.

Unit tests included.

Compatible with all old syntax, but includes a new one for getting managers.
```py
data_manager = knowledge.get_manager(DataManager)
```

### CODE REVIEW

File: sharpy/knowledges/knowledge_test.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+        assert data_manager is not None
+        assert isinstance(data_manager, TestDataManager)
+        assert isinstance(data_manager, DataManager)
+        assert data_manager.should_be_true()
+
+    @pytest.mark.asyncio
+    async def test_get_CustomManager_from_override(self):
+        knowledge = Knowledge()
+        knowledge.set_managers([CustomTestManager()])
```
Comment (merfolk) [inline sharpy/knowledges/knowledge_test.py:64]: Requiring a list to set a single manager is a bit clumsy.

I think that we should provide a method to set to set a single manager.

File: sharpy/knowledges/knowledge.py [hunk 2]
```diff
@@ -107,6 +108,16 @@ def set_managers(self, additional_managers: Optional[List[ManagerBase]]):
...
 
+    def get_manager(self, manager_type: Type[TManager]) -> TManager:
+        """
+
+        @param manager_type:
+        @return:
+        """
+        for manager in self.managers:
+            if issubclass(type(manager), manager_type):
+                return manager
```
Comment (merfolk) [inline sharpy/knowledges/knowledge.py:119]: I believe list is not the best data structure here.

Iterating the whole list is guaranteed to cause problems for a bot author that eg. uses the `get_manager()` method on every step for every manager. That may not be how `get_manager()` is meant to be used, but it's going to happen.

Comment (merfolk) [reply sharpy/knowledges/knowledge.py:119]: One way to resolve the potential performance problems is to at least warn a bot author about it in the method docstring. Which is included but empty. 😛

File: sharpy/knowledges/knowledge_test.py (hunk 1 above)
Comment (DrInfy) [reply sharpy/knowledges/knowledge_test.py:64]: The actual intended usage is supposed to be done in `KnowledgeBot` `configure_managers`. Testing was easier when using the exact `set_managers` method though. Should probably rename that to `_set_managers` and add a docstring warning.

File: sharpy/knowledges/knowledge_bot.py [hunk 3]
```diff
@@ -65,7 +65,7 @@ def _create_start_msg(self) -> str:
 
         return msg
 
-    async def chat_send(self, message: str):
+    async def chat_send(self, message: str, team_only: bool = False):
```
Comment (merfolk) [inline sharpy/knowledges/knowledge_bot.py:68]: This is not used any where?

Comment (DrInfy) [reply sharpy/knowledges/knowledge_bot.py:68]: Forgot to use it, fixed now. "Fixed" that before because PyCharm showed a warning there.

**Pull request merged**